cloud/workspace/workspace.go

```diff
@@ -120,33 +120,42 @@ var GetWorkspaceSelection = func(client astrocore.CoreClient, out io.Writer) (st
 	return selected.Id, nil
 }
 
-func Switch(id string, client astrocore.CoreClient, out io.Writer) error {
-	if id == "" {
-		_id, err := GetWorkspaceSelection(client, out)
+func Switch(workspaceNameOrID string, client astrocore.CoreClient, out io.Writer) error {
+	var wsID string
+	if workspaceNameOrID == "" {
+		id, err := GetWorkspaceSelection(client, out)
+		if err != nil {
+			return err
+		}
+
+		wsID = id
+	} else {
+		ws, err := GetWorkspaces(client)
 		if err != nil {
 			return err
 		}
+		for i := range ws {
+			if ws[i].Name == workspaceNameOrID || ws[i].Id == workspaceNameOrID {
+				wsID = ws[i].Id
+			}
+		}
 
-		id = _id
+		if wsID == "" {
+			return errors.Wrap(err, "workspace id/name is not valid")
```

Comment:
nit this seems a lil better
```suggestion
			return errors.Wrap(err, "workspace id/name is could not be found")
```

Reply:
@vandyliu this is grammatically incorrect :D .. You mean "workspace id/name could not be found".

Reply:
went with `workspace id/name could not be found`

Reply:
@vishwas-astro  🤦 oops, yep that is what i meant